### PR TITLE
Reworked network configuration

### PR DESCRIPTION
**Pull request by krzlabrdx**

### CODE REVIEW

File: apps/GatewayApi/Controllers/StatusController.cs [hunk 1]
```diff
@@ -118,7 +119,7 @@ public GatewayModel.NetworkConfigurationResponse NetworkConfiguration()
                 transactionProcessorPackage: wellKnownAddresses.TransactionProcessorPackage,
                 metadataModulePackage: wellKnownAddresses.MetadataModulePackage,
                 royaltyModulePackage: wellKnownAddresses.RoyaltyModulePackage,
-                accessRulesPackage: wellKnownAddresses.AccessRulesPackage,
+                accessRulesPackage: wellKnownAddresses.RoleAssignmentModulePackage,
```
Comment (PawelPawelec-RDX) [inline apps/GatewayApi/Controllers/StatusController.cs:122]: hmmm shouldn't we rename that field ?:P I know it's breaking change but...

File: src/RadixDlt.NetworkGateway.Abstractions/Network/INetworkConfigurationProvider.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+public interface INetworkConfigurationProvider
+{
+    public Task<NetworkConfiguration> GetNetworkConfiguration(CancellationToken token = default);
+}
+
+public sealed class NetworkConfigurationProvider : INetworkConfigurationProvider
+{
+    private readonly CoreApi.StatusApi _statusApi;
+    // private readonly IEnumerable<INetworkConfigurationReaderObserver>? _oservers; // TODO restore
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.Abstractions/Network/INetworkConfigurationProvider.cs:83]: do we have task for these todos to not forget about them?

**Review by PawelPawelec-RDX — CHANGES_REQUESTED** [review]

File: src/RadixDlt.NetworkGateway.Abstractions/Network/INetworkConfigurationProvider.cs (hunk 2 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.Abstractions/Network/INetworkConfigurationProvider.cs:83]: ok just reviewed second pr which addresses them. resolving.

File: apps/GatewayApi/Controllers/StatusController.cs (hunk 1 above)
Comment (krzlabrdx) [reply apps/GatewayApi/Controllers/StatusController.cs:122]: Yes, we definitely should! This change is part of separate PR (with xrd_usd_price)

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**